### PR TITLE
Get value from ArrayableString before applying markdown modifier

### LINKED ISSUE #3437: Markdown modifier cannot be applied to code fieldtype
## Bug Description
After updating from 3.0.46 to 3.1.1 I noticed a change in how a page was rendered. I've used the markdown modifier to parse a string, and it's no longer happening.

## How to Reproduce
Create a blueprint with a **code** fieldtype. Try to render this in a template with the markdown modifier.

## Extra Detail
fieldset
result
code
content

## Environment
Statamic 3.1.1 Pro
Laravel 8.26.1
PHP 7.3.12
No addons installed

**Install method (choose one):**
- Fresh install from `statamic/statamic`

### PR DESCRIPTION
**Pull request by arthurperton**
Fixes #3437.

### CODE REVIEW

File: src/Modifiers/CoreModifiers.php [hunk 1]
```diff
@@ -1227,7 +1228,10 @@ public function mailto($value, $params)
      */
     public function markdown($value, $params)
     {
-        if (! is_string($value)) {
+        if ($value instanceof ArrayableString) {
+            // NOTE: Could use 'instanceof Stringable' instead (and then cast to string) in PHP8+
+            $value = $value->value();
+        } elseif (! is_string($value)) {
             return $value;
         }
```
Comment (jasonvarga) [inline src/Modifiers/CoreModifiers.php:1236]: I think we might be able to just cast $value to a string rather than using these conditions.

**Review by jasonvarga — CHANGES_REQUESTED** [review]

File: src/Modifiers/CoreModifiers.php (hunk 1 above)
Comment (arthurperton) [reply src/Modifiers/CoreModifiers.php:1236]: Would be nice. In my tests I get an error though: 'Object of class ... could not be converted to string' when casting an object to string that doesn't implement `__toString()`.

Comment (jasonvarga) [reply src/Modifiers/CoreModifiers.php:1236]: Which test?

Comment (arthurperton) [reply src/Modifiers/CoreModifiers.php:1236]: Quick and dirty, I put something like this in phptester.net, the last statement will fail:

```php
<?php


class Willing
{
	public function __toString()
	{
		return 'okay!';
	}
}


class Unwilling
{
}


$value = new Willing;

$value = (string) $value;


$value = new Unwilling;

$value = (string) $value;


```

Comment (jasonvarga) [reply src/Modifiers/CoreModifiers.php:1236]: It doesn't make sense for someone to apply a markdown modifier to something that doesn't evaluate to a string.

Comment (arthurperton) [reply src/Modifiers/CoreModifiers.php:1236]: Yeah I agree. You think it should just fail loud then?

Comment (arthurperton) [reply src/Modifiers/CoreModifiers.php]: I'm guessing you do 🎲

**Review by jasonvarga — APPROVED** [review]
Awesome, thanks.

**Pull request merged**